### PR TITLE
Adds functions for setting certificate chain and private key from memory

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Comment by huitema** [comment]
Sorry, have to wait until tonight to review that

File: picoquic/picoquic.h [hunk 1]
```diff
@@ -259,6 +259,12 @@ void picoquic_free(picoquic_quic_t* quic);
 /* Set cookie mode on QUIC context when under stress */
 void picoquic_set_cookie_mode(picoquic_quic_t* quic, int cookie_mode);
 
+/* Set the TLS certificate chain(DER format) for the QUIC context. The context will take ownership over the certs pointer. */
+void picoquic_set_tls_certificate_chain(picoquic_quic_t* quic, ptls_iovec_t* certs, size_t count);
+
+/* Set the TLS private key(DER format) for the QUIC context. The caller is responsible for cleaning up the pointer. */
+int picoquic_set_tls_key(picoquic_quic_t* quic, const uint8_t* data, size_t len);
+
```
Comment (huitema) [inline picoquic/picoquic.h]: OK. The ptls_iovec_t is already defined in a forward reference, so we are good.

File: picoquic/tls_api.c [hunk 2]
```diff
@@ -641,8 +654,10 @@ void picoquic_master_tlscontext_free(picoquic_quic_t* quic)
             ctx->get_time = NULL;
         }
 
-        if (ctx->certificates.list != NULL) {
-            free(ctx->certificates.list);
+        free_certificates_list(ctx->certificates.list, ctx->certificates.count);
+
+        if (ctx->sign_certificate != NULL) {
+            ptls_openssl_dispose_sign_certificate((ptls_openssl_sign_certificate_t*)ctx->sign_certificate);
         }
```
Comment (huitema) [inline picoquic/tls_api.c]: Did we in fact have a memory leak before? Did you find out with a tool, or just by code review?

Comment (bkchr) [reply picoquic/tls_api.c]: Just by code review.

Comment (bkchr) [reply picoquic/tls_api.c]: And yes, I assume that before that would be a memory leak.

Comment (huitema) [reply picoquic/tls_api.c]: Do you happen to know a good open source tool to detect leaks?

Comment (bkchr) [reply picoquic/tls_api.c]: Yep, `valgrind` is the tool to use :)

**Pull request merged**